### PR TITLE
Add language dropdown menu

### PR DESCRIPTION
**Pull request by mbenadda**
## Purpose

As we have a React codebase that powers all of Richie's interactive interface elements, it makes sense for us to build the language selector using our frontend stack.

We're relying on downshift to build a dropdown menu with less code, and integrating it with the backend template.

## Proposal

- [x] create a `<LanguageSelector />` component
- [x] integrate it with our Django-CMS powered language picker template
- [x] make 2 different versions for the header and footer (white and dark background)

### CODE REVIEW

File: src/frontend/js/components/Root/index.tsx [hunk 1]
```diff
@@ -54,9 +57,22 @@ export const Root = ({ richieReactSpots }: RootProps) => {
...
-      // Get the incoming props to pass our component from the `data-props` attribute
+      let props: any = {};
+
+      // Get the props to pass our components from the `data-props-source` if
+      const dataPropsSource = element.getAttribute('data-props-source');
+      if (dataPropsSource) {
+        props = {
+          ...props,
+          ...JSON.parse(document.querySelector(dataPropsSource)!.textContent!),
+        };
```
Comment (lunika) [inline src/frontend/js/components/Root/index.tsx:68]:
```suggestion
        props = JSON.parse(document.querySelector(dataPropsSource)!.textContent!);
```

File: src/frontend/js/components/LanguageSelector/index.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      name: 'Select a language: Français',
+    });
+
+    userEvent.click(button);
+
+    screen.getByRole('listbox', { name: 'Select a language:' });
+    screen.getByRole('option', { name: 'Switch to Anglais' });
+    screen.getByRole('link', { name: 'Switch to Anglais' });
+    screen.getByRole('option', {
+      name: 'Switch to Français (currently selected)',
```
Comment (sampaccoud) [inline src/frontend/js/components/LanguageSelector/index.spec.tsx:39]: this is shown on hover and is not being translated

Comment (mbenadda) [reply src/frontend/js/components/LanguageSelector/index.spec.tsx:39]: It's expected not to be translated as we have not done the translations for this PR yet.

As for being shown on hover, this is a consequence of being the title of the link. Is it an issue? I think we're supposed to put this sort of thing as part of a link title when the link is "doing something", not just a regular link

Comment (sampaccoud) [reply src/frontend/js/components/LanguageSelector/index.spec.tsx:39]: Ah ok. Showing on hover is great. I was mentioning the absence of a translation.
You're right, we'll provide translations in the coming release.

**Review by lunika — APPROVED** [review]

File: src/frontend/js/components/LanguageSelector/_styles.scss [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      .language-selector--dark & {
+        color: white;
+
+        &:hover {
+          color: white;
+        }
+      }
+    }
+  }
+}
```
Comment (sampaccoud) [inline src/frontend/js/components/LanguageSelector/_styles.scss:84]: On mobile, for the footer version, I think half of the popin is showing outside of the screen.

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**